docs/DeveloperGuide.md

```diff
@@ -1,18 +1,120 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+The **Architecture Design** given above explains the high-level design of the App.
+Given below is a quick overview of each component. </br>
 
-## Product scope
-### Target user profile
+`Modman` is the main class of the application, and handles the app launch,
+initializing the appropriate classes to be used. </br>
+
+The rest of the app consists of four components:
+
+* Logic
+* UI
+* Storage
+* Data
+
+**How the four architecture components interact with each other and the ModMan class**
+
+The *Sequence diagram* below shows the interaction between components for when the user issues the command `add module CS2113T`
+
+![](uml/ArchitectureSequenceDiagram.png)
+
+The sections below give more details of each component.
+
+
+## Logic Component (Jianning)
+
+![logic](uml/ParserAndCommand.png)
+
+1. The `Parser` class is used to parse the user command.
+2. This returns a `Command` object which is executed in `Modman`.
+3. The command execution can affect the state of other components e.g. the sort command changes the order of assignments in `Module`.
+4. The `Command` object can also instruct the `Ui` to list and display information to the user.
+
+Given below is the Sequence Diagram for creating the corresponding `Command` object from user input via `Parser`. </br>
+The sequence diagram also acts as a reference frame for `getCommand`.
+
+![getCommand](uml/getCommand.png)
+
+## UI Component
+The UI Component consists of one class - `Ui` which provides all the functions
+required to print different kinds of messages on te console.
+
+1. `Ui` is class with a `Scanner` object that reads input from console.
+1. `Modman` has `Ui` object as an attribute which is instantiated
+   when we enter the program.
+1. `Ui` object reads input from screen and returns it which will then be passed to the Parser so that it can
+   parse the command.
+1. Depending on the parsed command, the corresponding
+   `execute()` function of the  `Command` abstract class
+   runs. It accepts `Ui` object as a parameter and calls the relevant `Ui` methods
+   that prints the messages.
+   <br>
+   <br>
+
+Given below is the Sequence Diagram for the Storage class. The storage class interacts with the database file, storing our data into the database or loading our data from the database. </br>
+## Storage Component (Bryan)
+![Storage](uml/StorageClassDiagram.png)
+
+1. The `Storage` class is used to interact with the database.txt file by calling loadData() at the start of the program and calling saveData() at the end of the program.
+2. The `Storage` class will collect the data of all classes that implements the Storable interface so that these data can be saved to the database.
```

Comment:
Since multiple classes are interacting with the Storage class, maybe it would be nice to include 1 small example of a class interacting with the Storage class with a sequence diagram?